packages/twenty-front/src/pages/settings/integrations/SettingsIntegrationEditDatabaseConnection.tsx

```diff
@@ -0,0 +1,247 @@
+import { useIsFeatureEnabled } from '@/workspace/hooks/useIsFeatureEnabled';
+import { RemoteTableStatus } from '~/generated-metadata/graphql';
+import { SettingsIntegrationDatabaseConnectionWrapper } from '~/pages/settings/integrations/SettingsIntegrationDatabaseConnectionWrapper';
+
+export const SettingsIntegrationEditDatabaseConnection = () => {
+  const { enqueueSnackBar } = useSnackBar();
+  const navigate = useNavigate();
+  const { databaseKey = '', connectionId = '' } = useParams();
+  const [integrationCategoryAll] = useSettingsIntegrationCategories();
+  const integration = integrationCategoryAll.integrations.find(
+    ({ from: { key } }) => key === databaseKey,
+  );
+
+  const isPostgresqlIntegrationEnabled = useIsFeatureEnabled(
+    'IS_POSTGRESQL_INTEGRATION_ENABLED',
+  );
+  const isIntegrationUpdateAvailable =
+    !!integration &&
+    databaseKey === 'postgresql' &&
+    isPostgresqlIntegrationEnabled;
+
+  const settingsIntegrationsPagePath = getSettingsPagePath(
+    SettingsPath.Integrations,
+  );
+
+  const { connection, loading } = useGetDatabaseConnection({
+    databaseKey,
+    connectionId,
+  });
+
+  useEffect(() => {
+    if (!isIntegrationUpdateAvailable || (!loading && !connection)) {
+      navigate(AppPath.NotFound);
+    }
+  }, [navigate, connection, loading, isIntegrationUpdateAvailable]);
+
+  const { tables } = useGetDatabaseConnectionTables({
+    connectionId,
+    skip: !connection,
+  });
+
+  const hasSyncedTables = tables.some(
+    (table) => table?.status === RemoteTableStatus.Synced,
+  );
+
+  const editConnectionSchema = getEditionSchema(databaseKey);
+  type SettingsIntegrationEditConnectionFormValues = z.infer<
+    typeof editConnectionSchema
+  >;
+
+  const formConfig = useForm<SettingsIntegrationEditConnectionFormValues>({
+    mode: 'onTouched',
+    resolver: zodResolver(editConnectionSchema),
+    defaultValues: useMemo(() => {
+      return getFormDefaultValues({ databaseKey, connection });
+    }, [databaseKey, connection]),
+  });
+
+  useEffect(() => {
+    formConfig.reset(getFormDefaultValues({ databaseKey, connection }));
+  }, [formConfig, databaseKey, connection]);
+
+  const { updateOneDatabaseConnection } = useUpdateOneDatabaseConnection();
+
+  if (!isIntegrationUpdateAvailable || !connection) return null;
+
+  const connectionName = getConnectionDbName({ integration, connection });
+
+  const canSave =
+    !hasSyncedTables &&
+    formConfig.formState.isDirty &&
+    formConfig.formState.isValid;
+
+  const handleSave = async () => {
+    const formValues = formConfig.getValues();
+    const dirtyFieldKeys = Object.keys(
+      formConfig.formState.dirtyFields,
+    ) as (keyof SettingsIntegrationEditConnectionFormValues)[];
+
+    try {
+      await updateOneDatabaseConnection({
+        ...formatValues({
+          databaseKey,
+          formValues: pick(formValues, dirtyFieldKeys),
+        }),
+        id: connectionId,
+      });
+
+      navigate(
```

Comment:
TODO in a later PR - when redirecting to this page, connection status is not updated

https://github.com/twentyhq/twenty/assets/51697796/aadd940a-0d88-4655-ab89-bae7b0e36869